Storage/Storage/Tools/StorageType+Extensions.swift

```diff
@@ -752,10 +752,18 @@ public extension StorageType {
 
     // MARK: - Meta Data
 
-    /// Retrieves the Stored Metadata.
+    /// Retrieves the Stored Metadata for Order.
     ///
-    func loadMetaData(siteID: Int64, metadataID: Int64) -> MetaData? {
+    func loadOrderMetaData(siteID: Int64, metadataID: Int64) -> MetaData? {
         let predicate = \MetaData.order?.siteID == siteID && \MetaData.metadataID == metadataID
         return firstObject(ofType: MetaData.self, matching: predicate)
     }
+
+    /// Retrieves the Stored Metadata for Order.
+    ///
+    func loadProductMetaData(siteID: Int64, metadataID: Int64) -> MetaData? {
+        let predicate = \MetaData.product?.siteID == siteID && \MetaData.metadataID == metadataID
+        return firstObject(ofType: MetaData.self, matching: predicate)
+    }
```

Comment:
Will `metadataID` be unique across products and orders? 

I wonder what happens when a product and an order have metadata with the same ID. Will we end up loading order's meta data for a product and vice versa?

Reply:
I discussed this here C03L1NF1EA3/p1725454480621309.

On the web, MetaData for both orders and products are sharing the same table in the db `wp_postmeta`. However, both in iOS and in Android, we use the same approach: MetaData with the same IDs are allowed because it's possible that in the future meta data with the same ID will be allowed. For the moment, even if the table is the same, the endpoints for updating/creating metadata are linked just to orders or just to products, so it is impossible to create shared metadata. And in any case, the relationship from MetaData to Order/Product in CoreData is the following:
```
Order -> many MetaData
Product -> many MetaData
MetaData -> one Order
MetaData -> one Product
```

Reply:
I also made some changes here to be sure to fetch MetaData connected with a specific product or order: 4d0859bcf785c0bce5f55a2e3a32f94a26096a72